src/main/webapp/scripts/forum-script.js

```diff
@@ -179,3 +194,37 @@ function collapseReplies(idHandler) {
   $('#' + elementId + ' .expand-button').css('display', 'inline-block');
   $('#' + elementId + ' .collapse-button').css('display', 'none');
 }
+
+/**
+ *  Onclick handler for a user searching to reload forum with search parameter
+ */
+function search() {
+  const search = $('#search-input').val();
+  $('#search-input').val('');
+  const id = -1;
+  const placeholder = $('#forum-placeholder');
+  expandForum(placeholder, id, search);
+}
+
+/**
+ *  Returns whether a word of the search is contained with in the text
+ *
+ *  @param {String} text contents of a question
+ *  @param {String} search contents of the user search
+ *
+ *  @return {Boolean} whether there is a word of the search within the text
+ */
+function containsSearch(text, search) {
+  const stopWords = ['a', 'to', 'and', 'how', 'the', 'when', 'what', 'why',
+    'what', 'where', 'or', 'do', 'can', 'use', 'i', 'you', 'my', 'your'];
+  const lowerText = text.toLowerCase();
+  const lowerSearch = search.toLowerCase();
+  let words = lowerSearch.split(' ');
+  words = words.filter( function(word) {
+    return ((!stopWords.includes(word)) && (lowerText.includes(word)));
```

Comment:
This is a good V0.5, but please research APIs to search data store.

Reply:
Because Datastore is No-SQL it seemed like there were no search capabilities, which is why I decided to manually code a search. But now I realize that doing it on the client side is not very efficient.  I will change this to either be on the server side or find a API after the other forum features are added.

Reply:
This will be in the next version

Reply:
I filed an issue for this.